### PR TITLE
Improve display of continuation lines with multiline errors

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/pytest-dev/pytest/issues/717.
Follow-up to https://github.com/pytest-dev/pytest/pull/1762.

### CODE REVIEW

**Comment by nicoddemus** [comment]
Hey @blueyed. thanks!

Looks good to me, perhaps though we should add a `CHANGELOG` entry for that as well.

File: _pytest/python.py [hunk 1]
```diff
@@ -1849,12 +1849,10 @@ def toterminal(self, tw):
...
             tw.line(tbline.rstrip())
         lines = self.errorstring.split("\n")
-        for line in lines:
-            if line == lines[0]:
-                prefix = 'E       '
-            else:
-                prefix = '        '
-            tw.line(prefix + line.strip(), red=True)
+        if lines:
+            tw.line('E       ' + lines[0].strip(), red=True)
```
Comment (nicoddemus) [inline _pytest/python.py]: Just noticed that we have this in `py._code.code.py`:

```python
class FormattedExcinfo(object):
    """ presenting information about failing Functions and Generators. """
    # for traceback entries
    flow_marker = ">"
    fail_marker = "E"
```

Could please change the code use those constants instead?

**Comment by tomviner** [comment]
LGTM :+1:

File: _pytest/python.py (hunk 1 above)
Comment (blueyed) [inline _pytest/python.py]: Done.  Not sure though if that is really good like that.
After all it should be some more generic method probably for this kind of messages?

**Comment by blueyed** [comment]
Please don't merge this (there are fixup-commits meant to be squashed), but squash-merge it (when using the GitHub UI), or let me know and I'll squash and push it.

**Comment by nicoddemus** [comment]
Thanks @blueyed and @tomviner!

**Comment by blueyed** [comment]
FWIW: #1762 should have been made against the features branch probably.  There are huge conflicts now, since it was split in 8c49561.

**Comment by blueyed** [comment]
@nicoddemus 
In case you don't know about rerere already: https://git-scm.com/blog/2010/03/08/rerere.html (outdated, better check the man page probably).

**Comment by nicoddemus** [comment]
Thanks for the heads up @blueyed!

**Comment by tomviner** [comment]
@blueyed said:
> FWIW: #1762 should have been made against the features branch probably. 

That's probably by bad, I was advising new contributors at the Europython sprint 😬

Looking again at https://github.com/pytest-dev/pytest/issues/717 (which #1762 attempts to fix) it's more of a "bring into line" (with other traceback output), than strictly a "bugfix". So I guess default to features branch for these kind of tickets?

**Comment by nicoddemus** [comment]
@tomviner not sure, that's such a small change in output, I think it is OK for it to go to master. Others may have other opinions though. :grin:

**Comment by The-Compiler** [comment]
I'm still wondering what crazy output parsing people do (like for IDEs) though, and how much we break even with small output changes.

**Comment by tomviner** [comment]
@The-Compiler I'm sure they do all sorts. But if we have faith in the longevity of pytest, perhaps being more consistent now will help _future crazy output parsers_ ;-)

**Comment by The-Compiler** [comment]
@tomviner Sure, but I think that's a good reason to have any changes to the output in `features` :wink:

**Comment by tomviner** [comment]
@The-Compiler see what you mean. So really, for patches that alter the pytest interface, only changes that were obviously broken before should be aimed at master.

**Pull request merged**